molgenis-app-gavin2/src/main/java/org/molgenis/app/gavin/GavinController.java

```diff
@@ -0,0 +1,191 @@
+package org.molgenis.app.gavin;
+
+import static java.lang.String.format;
+import static java.util.Arrays.asList;
+import static java.util.Objects.requireNonNull;
+import static org.molgenis.app.gavin.GavinController.URI;
+import static org.molgenis.app.gavin.meta.GavinRunMetadata.GAVIN_RUN;
+import static org.molgenis.data.file.model.FileMetaMetaData.FILE_META;
+import static org.molgenis.data.importer.ImportRunMetaData.STATUS;
+import static org.springframework.http.MediaType.APPLICATION_OCTET_STREAM_VALUE;
+
+import java.io.File;
+import java.io.IOException;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.stream.Stream;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.molgenis.app.gavin.meta.GavinRun;
+import org.molgenis.data.DataService;
+import org.molgenis.data.file.FileStore;
+import org.molgenis.data.file.model.FileMeta;
+import org.molgenis.jobs.model.JobExecution.Status;
+import org.molgenis.security.core.runas.RunAsSystem;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.io.FileSystemResource;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.multipart.MultipartFile;
+
+@RestController
+@RequestMapping(URI)
+public class GavinController {
+  private static final Logger LOG = LoggerFactory.getLogger(GavinController.class);
+
+  static final String GAVIN = "/api/gavin";
+  static final String URI = GAVIN;
+  private final GavinService gavinService;
+  private final FileStore fileStore;
+  private final DataService dataService;
+
+  private static final Duration RUN_EXPIRATION_TIME = Duration.ofDays(1);
+
+  GavinController(GavinService gavinService, FileStore fileStore, DataService dataService) {
+    this.gavinService = requireNonNull(gavinService);
+    this.fileStore = requireNonNull(fileStore);
+    this.dataService = requireNonNull(dataService);
+  }
+
+  @RunAsSystem
+  @PostMapping(value = "/upload")
+  public ResponseEntity<String> upload(
+      @RequestParam(value = "file") MultipartFile inputFile, HttpServletRequest httpServletRequest)
```

Comment:
This file will be written to the temp folder of the servlet container if the size is larger than ? MB. Is this acceptable?

Reply:
According to Joeri this is acceptable for now